### PR TITLE
Fix references to `this` broken by linting

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Remaining references to `self` are necessary to avoid linter errors; this reverts those references that broke functionality (and also happened to not be linter errors).

### CODE REVIEW

File: src/app/scripts/cac/search/cac-typeahead.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ CAC.Search.Typeahead = (function ($) {
             self.events.trigger('cac:typeahead:selected', [typeaheadKey, suggestion]);
         } else {
             CAC.Search.Geocoder.search(suggestion.text, suggestion.magicKey).then(function (location) {
-                self.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
+                this.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
```
Comment (notthatbreezy) [inline src/app/scripts/cac/search/cac-typeahead.js]: Still get an error here when trying to geocode

```
Uncaught TypeError: Cannot read property 'trigger' of undefined cac-typahead.js:60
```

**Comment by flibbertigibbet** [comment]
Okay, all `this` references have been restored; defining `self` apparently does not work in all contexts.  JSHint will now ignore the strict check for `this` references.

File: src/app/scripts/cac/search/cac-typeahead.js [hunk 2]
```diff
@@ -54,10 +53,10 @@ CAC.Search.Typeahead = (function ($) {
         var typeaheadKey = $(event.currentTarget).data('typeahead-key') || defaultTypeaheadKey;
 
         if (dataset === 'currentlocation') {
-            self.events.trigger('cac:typeahead:selected', [typeaheadKey, suggestion]);
+            this.events.trigger('cac:typeahead:selected', [typeaheadKey, suggestion]);
         } else {
             CAC.Search.Geocoder.search(suggestion.text, suggestion.magicKey).then(function (location) {
-                self.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
+                this.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
```
Comment (notthatbreezy) [inline src/app/scripts/cac/search/cac-typeahead.js]: This (no pun intended) still throws an error -- the whole function needs to be reverted so it should be

```javascript
    function onTypeaheadSelected(event, suggestion, dataset) {
        var self = this;
        var typeaheadKey = $(event.currentTarget).data('typeahead-key') || defaultTypeaheadKey;

        if (dataset === 'currentlocation') {
            self.events.trigger('cac:typeahead:selected', [typeaheadKey, suggestion]);
        } else {
            CAC.Search.Geocoder.search(suggestion.text, suggestion.magicKey).then(function (location) {
                self.events.trigger('cac:typeahead:selected', [typeaheadKey, location]);
            });
        }
    }
```

**Comment by flibbertigibbet** [comment]
I saw no console errors in testing typeahead; reverted function anyways.

**Comment by notthatbreezy** [comment]
You have to hit geocode, there are no errors just from the typahead.

**Comment by flibbertigibbet** [comment]
Okay... so are you fine with this now?

**Comment by notthatbreezy** [comment]
+1

**Pull request merged**